client/src/main/java/org/asynchttpclient/netty/ssl/DefaultSslEngineFactory.java

```diff
@@ -26,13 +26,11 @@
 
 public class DefaultSslEngineFactory extends SslEngineFactoryBase {
 
-    private final SslContext sslContext;
+    private volatile SslContext sslContext;
 
-    public DefaultSslEngineFactory(AsyncHttpClientConfig config) throws SSLException {
-        this.sslContext = getSslContext(config);
-    }
+    public DefaultSslEngineFactory() { }
 
-    private SslContext getSslContext(AsyncHttpClientConfig config) throws SSLException {
+    private SslContext buildSslContext(AsyncHttpClientConfig config) throws SSLException {
```

Comment:
I left this code where it was in the source but if you don't mind the reshuffling, I'll move it down to the bottom with the other private stuff